### PR TITLE
Implement array_split with cython and optimize it

### PR DESCRIPTION
**Pull request by unnonouno**
I optimized array_split method with Cython. The original implementation calls `__getitem__` method many times. I removed this call and made views manually, and it is simple enough.

related to #1492

### CODE REVIEW

**Comment by unnonouno** [comment]
This is benchmark result. Only about 1.0 ms in 5.10 ms is consumed in `cupy.array_split` method. This method generates a lot of `Variable` object as its result. So, the bottleneck is now in `Variable`.

```
CPU split_axis ... finished in 5.26ms
CPU concat ... finished in 3.54ms
copy to GPU ... finished in 1.38ms
GPU split_axis ... finished in 5.10ms
GPU concat ... finished in 18.93ms
copy to CPU ... finished in 2.14ms
```

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -1810,6 +1810,52 @@ cpdef ndarray rollaxis(ndarray a, Py_ssize_t axis, Py_ssize_t start=0):
...
+    if -ndim > axis or ndim <= axis:
+        raise IndexError('Axis exceeds ndim')
+    if axis < 0:
+        axis += ndim
+    size = ary._shape[axis]
+
+    if numpy.isscalar(indices_or_sections):
+        each_size = (size - 1) // indices_or_sections + 1
+        indices = [i * each_size
+                   for i in six.moves.range(1, indices_or_sections)]
```
Comment (okuta) [inline cupy/core/core.pyx]: Please use `range` instead of `six.moves.range`.

Comment (unnonouno) [inline cupy/core/core.pyx]: Ah, ok

**Comment by okuta** [comment]
thanks

**Pull request merged**